comp/core/hostname/remotehostnameimpl/hostname.go

```diff
@@ -0,0 +1,133 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+// Package remotehostnameimpl provides a function to get the hostname from core agent.
+package remotehostnameimpl
+
+import (
+	"context"
+	"time"
+
+	"github.com/avast/retry-go/v4"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	pbgo "github.com/DataDog/datadog-agent/pkg/proto/pbgo/core"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+	"github.com/DataDog/datadog-agent/pkg/util/grpc"
+	"github.com/DataDog/datadog-agent/pkg/util/hostname"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+
+	"github.com/DataDog/datadog-agent/comp/core/hostname/hostnameinterface"
+	cache "github.com/patrickmn/go-cache"
+	"go.uber.org/fx"
+)
+
+const (
+	defaultExpire = 15 * time.Minute
```

Comment:
Looks like these values are defaults but cannot be overridden, should these be configurable? Same goes for `maxAttempts`. That would mean, of course, we'd need a dependency with the configuration component.

Reply:
I would not to add configuration for this. Customers should worry about it.  I don't think it currently configurable in any agent